crit/mempages.go

```diff
@@ -193,3 +194,33 @@ func (mr *MemoryReader) GetShmemSize() (int64, error) {
 
 	return size, nil
 }
+
+// SearchPattern searches for a regular expression pattern in the process memory pages.
+// If the pattern is found, it returns the absolute address of the first byte of the pattern.
+// If the pattern is not found, an error is returned.
+func (mr *MemoryReader) SearchPattern(pattern string) (uint64, error) {
+	regexPattern, err := regexp.Compile(pattern)
+	if err != nil {
+		return 0, err
+	}
+
+	for _, entry := range mr.pagemapEntries {
+		startAddr := entry.GetVaddr()
+		endAddr := startAddr + uint64(entry.GetNrPages())*uint64(mr.pageSize)
+
+		// Read the entire page content
+		pageContent, err := mr.GetMemPages(startAddr, endAddr)
+		if err != nil {
+			return 0, err
+		}
+
+		// Search for the pattern using the regular expression
+		if match := regexPattern.FindIndex(pageContent.Bytes()); match != nil {
```

Comment:
Would it make sense to use `FindAllIndex` instead? Ideally we would want to return all matches in the memory, rather than just the first one.